### ISSUE TITLE
Add sway to nvim-treesitter and deprecate syntax file

### ISSUE THREAD

**Issue by sdankel** [FuelLabs/sway.vim#11]
We should deprecate the syntax highlighting file in this repo and instead add sway to [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter?tab=readme-ov-file#supported-languages). That way we only need to keep the [tree-sitter-sway](https://github.com/FuelLabs/tree-sitter-sway) grammar updated.

**Comment by yehia67** [comment]
Hi @sdankel!
Can I help?

**Comment by sdankel** [comment]
@yehia67 Please do! The package is published at https://www.npmjs.com/package/tree-sitter-sway.

**Comment by yehia67** [comment]
On it 🏃 
My solution:
1. Delete the [syntax/sway.vim](https://github.com/FuelLabs/sway.vim/blob/master/syntax/sway.vim) file
2. Create `init.vim` file.
3. Add `nvim-treesitter` to the `init.vim` file
```
Plug 'nvim-treesitter/nvim-treesitter', {'do': ':TSUpdate'}
```
Wdyt @sdankel ?

**Comment by sdankel** [comment]
We want to make sway a supported language in `nvim-treesitter`. See https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md

**Comment by yehia67** [comment]
Yeah, You successfully supported it here [tree-sitter-sway](https://github.com/FuelLabs/tree-sitter-sway). 
So I need to find a way to install [tree-sitter-sway](https://github.com/FuelLabs/tree-sitter-sway) syntax on the `syntax/sway.vim` instead of this [highlighting syntax block](https://github.com/FuelLabs/sway.vim/blob/master/syntax/sway.vim#L211-L81), yes? (Just to confirm if I understood the requirements 🙏)

**Comment by sdankel** [comment]
The flow for the user would be that they simply install [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter?tab=readme-ov-file#quickstart), and then run `:TSInstall sway` to install the sway syntax. They would no longer need to copy a `syntax/sway.vim` from this repo.

To do this, we'll need to submit a PR to that repo to add the tree-sitter-sway parser. Here are example PRs:
https://github.com/nvim-treesitter/nvim-treesitter/pulls?q=is%3Apr+add+label%3A%22new+language%22

**Comment by yehia67** [comment]
Ah, that makes perfect sense. Sorry for the confusion. I've created a draft pull request (PR) on my forked repository, which you can find [here](https://github.com/yehia67/nvim-treesitter/pull/1). Please let me know if there's anything else you need me to add; I would be happy to do so.

Additionally, I've designated you as the maintainer of the language in this [section](https://github.com/yehia67/nvim-treesitter/blob/adc66b657e387d7d4d3270496bd144c77871ce67/README.md?plain=1#L395) of the README. If you would prefer that I tag someone else or mention myself instead, please let me know.

**Comment by sdankel** [comment]
The PR looks good to me, and thanks for adding me as the maintainer. Once it's merged and we can verify it, the next steps will be removing the syntaxes file from this repo and updating the README. Thank you for contributing!

**Comment by yehia67** [comment]
Thank you for your support 🙏  
I created the PR👇 
https://github.com/nvim-treesitter/nvim-treesitter/pull/6251

**Comment by yehia67** [comment]
Hi @sdankel,
The parser compiler for the queries is giving some "Invalid capture" errors, I fixed some of the errors with his [commit](https://github.com/nvim-treesitter/nvim-treesitter/pull/6251/commits/7377d77f00ec6da085acb5ffa7d66d3d7a5784fd) I am not 100% sure about my fixes tbh as I just copied from other projects, May you please point me to any resource for fixing the parser issue and I can work on it

**Comment by yehia67** [comment]
The maintainer of transmitter told me how to fix the queries, By using syntaxes of the Contribue.md I will check and keep you updated
